code/Controllers/CMSMain.php

```diff
@@ -974,19 +999,25 @@ public function Breadcrumbs($unlinked = false)
     public function SiteTreeHints()
     {
         $classes = SiteTree::page_type_classes();
-
-        $cacheCanCreate = array();
-        foreach ($classes as $class) {
-            $cacheCanCreate[$class] = singleton($class)->canCreate();
+        $memberID = Security::getCurrentUser()->ID;
+        $cache = $this->hintsCache;
+        $classList = implode(',', $classes);
+
+        // Cache the results of canCreate()
+        $permissionsCacheKey = md5(implode('_', [$memberID, $classList]));
```

Comment:
$classlist isn't a variable, it's a fact (fixed) for each project. I know it was there earlier but it's probably redundant, now that we have `flush` to clear (should be done when adding new classes).